lib/pxlogger.js

```diff
@@ -9,20 +9,25 @@ class PxLogger {
         this.appId = '';
     }
 
+    setCustomLogger(customLogger) {
```

Comment:
Since we are setting the logger in the init function below, do we need a separate function to do this? Seems like the call to this is duplicative of the init function?

Reply:
yeah, not the biggest fan of this either, but this is why I did it this way:

- The logger gets initiated [here](https://github.com/PerimeterX/perimeterx-node-core/blob/master/lib/pxenforcer.js#L23), and takes in an instance of PxConfig.
- But before this happens, logger gets passed in the constructor of PxConfig [here](https://github.com/PerimeterX/perimeterx-node-core/blob/master/lib/pxenforcer.js#L20), then logs some things in `mergeConfigFileParams` [here](https://github.com/PerimeterX/perimeterx-node-core/blob/master/lib/pxconfig.js#L52), which gets called in pxconfig constructor.
- So if we want those error logs via custom logger, we need the custom logger to be set before we call `mergeParams` [here](https://github.com/PerimeterX/perimeterx-node-core/blob/master/lib/pxconfig.js#L11)

Like I said, I don't like this either, so open for suggestions.